### PR TITLE
docs(website): help troubleshoot issues with multiple parsers

### PR DESCRIPTION
**Pull request by jfbloom22**
This might not be a perfect fit for the typescript-eslint docs, since this is specific to Vue.  However, I expect the issue effects other custom parsers as well. Including multiple parsers like this is counter intuitive because it seems like a mistake to include two "parser"s.  It would have been a huge help to me to have found this in the troubleshooting guide.  

more info:
https://eslint.vuejs.org/user-guide/#how-to-use-a-custom-parser

devs running into this issue:
https://github.com/vuejs/vue-eslint-parser/issues/45#issuecomment-988625895 https://stackoverflow.com/questions/66597732/eslint-vue-3-parsing-error-expected-eslint

## PR Checklist

- [ ] Addresses an existing open issue: fixes #000
- [ ] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [ ] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

### CODE REVIEW

File: docs/linting/TROUBLESHOOTING.md [hunk 1]
```diff
@@ -241,3 +241,18 @@ However this involves a _lot_ of computations across every single token in a fil
 Across a large codebase, these can add up, and severely impact performance.
 
 We recommend not using this rule, and instead using a tool like [`prettier`](https://www.npmjs.com/package/prettier) to enforce a standardized formatting.
+
+## I am running into errors when parsing Typescript in my .vue files
```
Comment (JoshuaKGoldberg) [inline docs/linting/TROUBLESHOOTING.md:245]: Nit:

```suggestion
## I am running into errors when parsing TypeScript in my .vue files
```

File: docs/linting/TROUBLESHOOTING.md [hunk 2]
```diff
@@ -241,3 +241,18 @@ However this involves a _lot_ of computations across every single token in a fil
...
+```diff
+- "parser": "@typescript-eslint/parser",
++ "parser": "vue-eslint-parser",
+  "parserOptions": {
++     "parser": "@typescript-eslint/parser",
+      "sourceType": "module"
+  }
+```
+
+The `parserOptions.parser` option can also specify an object to specify multiple parsers. See [vue-eslint-parser README](https://github.com/vuejs/vue-eslint-parser#readme) for more details.
```
Comment (JoshuaKGoldberg) [inline docs/linting/TROUBLESHOOTING.md:258]: Nit:

```suggestion
The `parserOptions.parser` option can also specify an object to specify multiple parsers. See the [vue-eslint-parser README](https://github.com/vuejs/vue-eslint-parser#readme) for more details.
```

File: docs/linting/TROUBLESHOOTING.md (hunk 1 above)
Comment (JoshuaKGoldberg) [inline docs/linting/TROUBLESHOOTING.md:245]: A previous entry on the page mentions Vue already: _I use a framework (like Vue) that requires custom file extensions, ..._. What do you think about moving this entry to just after that other one, to group Vue entries together?

Reasoning: if someone is reading about Vue, they probably want to see both FAQs.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Yeah this seems pretty reasonable 🙂 I think it's fine to not have a backing issue. Just a few small suggestions!

**Comment by JoshuaKGoldberg** [comment]
Ping @jfbloom22, do you still have time to work on this? No worries if not - just don't want to keep this pending too long. 🙂

File: docs/linting/TROUBLESHOOTING.md (hunk 1 above)
Comment (jfbloom22) [reply docs/linting/TROUBLESHOOTING.md]: thanks!

File: docs/linting/TROUBLESHOOTING.md (hunk 2 above)
Comment (jfbloom22) [reply docs/linting/TROUBLESHOOTING.md]: thanks

File: docs/linting/TROUBLESHOOTING.md (hunk 1 above)
Comment (jfbloom22) [reply docs/linting/TROUBLESHOOTING.md]: good idea!

**Comment by jfbloom22** [comment]
thanks for the suggestions!  I have updated the PR.

File: docs/linting/TROUBLESHOOTING.md [hunk 3]
```diff
@@ -86,6 +86,21 @@ module.exports = {
...
+```diff
+- "parser": "@typescript-eslint/parser",
++ "parser": "vue-eslint-parser",
+  "parserOptions": {
++     "parser": "@typescript-eslint/parser",
+      "sourceType": "module"
+  }
+```
+
+The `parserOptions.parser` option can also specify an object to specify multiple parsers. See the [vue-eslint-parser README](https://github.com/vuejs/vue-eslint-parser#readme) for more details.
```
Comment (JoshuaKGoldberg) [inline docs/linting/TROUBLESHOOTING.md:102]: The docs link you chose to link in the PR description is also really good: https://eslint.vuejs.org/user-guide/#usage. If that's the right place to send folks, maybe we should send them there instead?

```suggestion
The `parserOptions.parser` option can also specify an object to specify multiple parsers. See the [vue-eslint-parser usage guide](https://eslint.vuejs.org/user-guide/#usage) for more details.
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Awesome, thanks for this - great info and great summary!

Just suggesting one change but I'm not a Vue user myself so I'll defer to you on it. Once the build passes this is good to go! 🚀

**Review by JoshuaKGoldberg — APPROVED** [review]
Found this sitting on the queue, let's get it merged! 💪

**Pull request merged**